common/src/chain/transaction/signature/inputsig/authorize_pubkey_spend.rs

```diff
@@ -66,4 +66,221 @@ pub fn sign_pubkey_spending(
     Ok(AuthorizedPublicKeySpend::new(signature))
 }
 
-// TODO: tests
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::{
+        address::pubkeyhash::PublicKeyHash,
+        chain::{
+            signature::{
+                inputsig::{InputWitness, StandardInputSignature},
+                sighashtype::SigHashType,
+                signature_hash,
+            },
+            Destination, Transaction, TransactionCreationError, TxInput, TxOutput,
+        },
+        primitives::{Amount, Id},
+    };
+    use crypto::key::{KeyKind, PrivateKey, PublicKey};
+
+    fn generate_unsigned_tx(
```

Comment:
When you generate a transaction with just one output, it's not really possible to test the difference between ALL and SINGLE, because they'll be equivalent.